### PR TITLE
fix : i18n

### PR DESCRIPTION
**Pull request by kabir-afk**
### June 13
  - [refactor : storeParams now working properly](https://github.com/lingua-libre/SignIt/commit/6304977f2f020fbf4f7f5c09ad80431a03d68b44) - Used array destructuring to fetch the arguments properly. With this `storeParam` message now works properly and invokes the `storeParam` function inside `sw.js` and `background-script.js` . 
  I think this addresses #87 . Although #56 still persists.
  - [chore: passing message to execute i18n natively inside sw/bg.js](https://github.com/lingua-libre/SignIt/commit/94aaa8d7df34bf4ace617acf987f1e563ecdd18d) - This is the **highlight of this PR**. I figured it was better to pass messages from popup/other content scripts to execute `banana.i18n` directly inside `sw.js/background-script.js` and then send the response acccordingly.
   Pros and cons of what we were doing earlier vs now :
    - ### EARLIER [EDIT: Chrome native's `browser.i18n` API]
      - **Pros** :
        - It worked as expected and wasn't too asynchronous in nature, relied on local storage in the name of asynchronous nature.
      - **Cons** : 
        - **Verbosity changed depending on the context of usage :** For example - [SignItVideosGallery](https://github.com/lingua-libre/SignIt/blob/63d6c055cbb590afa6661d895a5d0a4704810b2c/SignItVideosGallery.js#L24C2-L37C5) and [SignItCoreContent](https://github.com/lingua-libre/SignIt/blob/63d6c055cbb590afa6661d895a5d0a4704810b2c/SignItCoreContent.js#L2C3-L3C58). In both the files different implementations of the same functions exist.
        - **Repetitve in nature :** If some other content script or constructor function popped , we'd have to copy/paste it again in order to make it work.
        - **Buggy UI :** While changing UI language , Popup had to be toggled off/on in order to see the changes and similar  had to be done for the modal. This might have been due to `SignItCoreContent` constructor not initializing timely , rather than a local storage issue. Although I am not sure why this occured as upon being passed with chosen arguements (user's preferred language ), it was logged correctly but rendered previous language.
    - ### NOW [EDIT: Wikimedia `banana.i18n` sent up and processed in sw.js/background.js]
      - **Pros** :
        - **Authenticity:** leverages the original library , hence maintaining the authenticity.
        - **Message passing:** only needs a message to execute.
        - **Little to no verbosity:** Unlike our own i18n , this works the same in all the contexts and we dont have to redefine it as per our preference every time.
       - **Cons** :
         - **Requires an async environment:** As we are passing messages , we are returned with a promise which needs to be resolved whenever the function is called. This can become problematic when working with constructors. This is dealt with in [fix : implemented the same message passing functionality as in popup](https://github.com/lingua-libre/SignIt/commit/d33217a0b37c7fa1ff8c4d543c10bcd159c333b7) by using an async prototype function named `init`, although not the best workaround.
         - **Irregualar html parsing:** While the library `banana.i18n` addresses its own , if not added as html inside the script, will render the string as is.
         

With this #84 can be closed now.
         
- [reverting to banana.i18n](https://github.com/lingua-libre/SignIt/commit/569136841b36e2536419b1c3b3008d7b348a645f) - Since we are now using `banana.i18n` , replacing `browser.i18n` API as well as its messages.
- [fix : implemented the same message passing functionality as in popup](https://github.com/lingua-libre/SignIt/commit/d33217a0b37c7fa1ff8c4d543c10bcd159c333b7) - This tries to achieve the same , but since its a constructor , promises had to be resolved first inside an async prototype, then inserted separately inside it using jQuery DOM manipulation. The only issue is that whenever it is called `init` has to be invoked as shown in [refactor : updated how SignItCoreContent() is called](https://github.com/lingua-libre/SignIt/commit/53c6bacad833143170757d9d0cd9c427a05c5244).
- [refactor: implemented the same message passing i18n functionality](https://github.com/lingua-libre/SignIt/commit/4407699ea6feaa2c7e28eac45063428065256944) - Same thing as earlier.

### June 14

- [refactor: used HTML instead of text for videosPanelNoVideoEmpty](https://github.com/lingua-libre/SignIt/commit/9dfaa56cf16e40c2022caeb48b511cc03ac1098f) - As mentioned above in the cons, earlier when used `.text()` , rendered `<br/>` as is. But replacing it with `.html()` resolved it.
- [storing banana and sourceMap no longer needed](https://github.com/lingua-libre/SignIt/commit/5793669e1cf260396872ee69fbe60938fdded2f0) - we don't need to store this inside `loadi18Localization` now as we no longer rely on it.

### CODE REVIEW

**Comment by kabir-afk** [comment]
**NOTE:** Just in case you get `HTTP Status Code 502 : Method not allowed`. comment out the 
`signLanguages = await getSignLanguagesWithVideos();` and `records = await getAllRecords( signLanguage );` line in both `sw.js` and `background-script.js`. I haven't thoroughly tested the changes in FF, so please let me know if there are some discrepancies.

File: SignItCoreContent.js [hunk 1]
```diff
@@ -62,9 +51,42 @@ var SignItCoreContent = function (locale,mapi18n) {
...
+        const translations = await Promise.all([
+          banana.i18n("si-panel-videos-title"),
+          banana.i18n("si-panel-videos-empty"),
+          banana.i18n("si-panel-definitions-title"),
+          banana.i18n("si-panel-videos-contribute-label"),
+          banana.i18n("si-panel-definitions-wikt-iso"),
+          banana.i18n("si-panel-definitions-wikt-pointer"),
+          banana.i18n("si-panel-definitions-empty") // May need a different key for error message
+        ]);
+        const [videosPanelNoVideoTitle, videosPanelNoVideoEmpty, definitionsPanelTitle, contributeButtonLabel, wiktIso, wiktPointer, definitionsEmpty] = translations;
```
Comment (hugolpz) [inline SignItCoreContent.js:71]: I don't see raison for the negation `No`, nor why `videos` appear at begging while the general convention so far is from wider to specific such as `SI > Panel > Videos|Definitions > [message]`. 

Variable names suggested : 
- `videosPanelNoVideoTitle` -> `panelVideosTitle`
- `videosPanelNoVideoEmpty` --> `panelVideosEmpty`
- `definitionsPanelTitle` --> `panelDefinitionsTitle`
-  `contributeButtonLabel` --> `panelVideosContributeLabel`
- `wiktIso`
- `wiktPointer`
- `definitionsEmpty` - > `panelDefinitionsEmpty`

I believe the prefix `panel` with the camelCase ensure a WebExt-wide uniqueness.

**Comment by hugolpz** [comment]
> This is the highlight of this PR. I figured it was better to pass messages from popup/other content scripts to execute banana.i18n directly inside sw.js/background-script.js and then send the response acccordingly.

Thank @kabir-afk for figuring that out !

File: SignItCoreContent.js [hunk 2]
```diff
@@ -1,49 +1,38 @@
-var SignItCoreContent = function (locale,mapi18n) {
-  const sourceMap = new Map(mapi18n);
-  banana = { i18n: (msg) => sourceMap.get(locale)[msg] };
-  console.log("Passed trough ! :", locale);
-  console.log("SignItCoreContent.js",banana );
-  // let hlwa = browser.i18n.getMessage("si-panel-videos-title"); 
-  // console.log("hlwa = ",hlwa); 
+var SignItCoreContent = function () {
+  console.log("SignItCoreContent.js");
```
Comment (Ishan-Saini) [inline SignItCoreContent.js:2]: Why are we logging on console ?

File: SignItCoreContent.js [hunk 3]
```diff
@@ -1,49 +1,38 @@
...
 		<div class="signit-modal-container">
 			<h1></h1>
 			<div class="signit-modal-content">
         <div class="signit-panel-videos">
           <div class="signit-panel-videos signit-novideo">
-            <h2>
-            ${ browser.i18n.getMessage("si_panel_videos_title") }</h2>
-            ${ browser.i18n.getMessage("si_panel_videos_empty") }<br><br>
+            <h2></h2>
+            <p></p>
```
Comment (Ishan-Saini) [inline SignItCoreContent.js:10]: Do we need these empty tags?

File: SignItVideosGallery.js [hunk 4]
```diff
@@ -17,42 +17,32 @@ var SignItVideosGallery = function ( container ) {
...
+	
 	this.$videoContainer.empty();
 	this.currentIndex = 0;
-
 	for ( i = 0; i < files.length; i++ ) {
 		filename = files[ i ].filename,
 		url = `https://commons.wikimedia.org/wiki/File:${ filename.split( '/' ).pop()}`,
 		speaker = files[ i ].speaker,
 		total = files.length;
+		console.log(await banana.i18n("si-panel-videos-gallery-attribution",url, speaker, i+1, total));
```
Comment (Ishan-Saini) [inline SignItVideosGallery.js:41]: This is not needed in my opinion, might be a good idea to remove this.

File: SignItCoreContent.js (hunk 2 above)
Comment (hugolpz) [reply SignItCoreContent.js:2]: This is legacy, for when i inspected, to know to which file the following logs were related to. Ca be removed if needed

File: SignItCoreContent.js (hunk 1 above)
Comment (kabir-afk) [reply SignItCoreContent.js:71]: Guess I'll change it then

File: SignItVideosGallery.js (hunk 4 above)
Comment (kabir-afk) [reply SignItVideosGallery.js:41]: yea...was doing that to test if the links were being parsed properly. Due to error 502 , videos weren't available and hence had to log them seperately in order to test it. Will remove it in the next PR.

**Pull request merged**